src/amber/router/context.cr

```diff
@@ -15,7 +15,6 @@ class HTTP::Server::Context
   def initialize(@request : HTTP::Request, @response : HTTP::Server::Response)
     @router = Amber::Router::Router.instance
     parse_params
-    upgrade_request_method!
     @route = router.match_by_request(@request)
```

Comment:
Is the route going to match since the method override doesn't happen until in the pipe?